### PR TITLE
Refactor handle creation methods

### PR DESCRIPTION
**Pull request by csilvestrini**
Introduces HandleSpec, which collects all of the configuration options for starting a handle. This reduces a bunch of duplicated code, and will make it much easier to introduce Reference Handles in a follow-up PR.

### CODE REVIEW

**Comment by csilvestrini** [comment]
cl/303668958

File: java/arcs/core/host/EntityHandleManager.kt [hunk 1]
```diff
@@ -61,85 +63,68 @@ class EntityHandleManager(
...
+            spec.baseName
+        ).toString()
+        val entityPreparer = EntityPreparer<T>(
+            handleName,
+            idGenerator,
+            spec.entitySpec.SCHEMA,
+            ttl,
+            time
+        )
+        return when (spec.containerType) {
```
Comment (jblebrun) [inline java/arcs/core/host/EntityHandleManager.kt]: this `when` is pretty long; splitting `Singleton` and `Collection` into private helper functions might be nice?

File: java/arcs/core/entity/Handle.kt [hunk 2]
```diff
@@ -25,6 +27,20 @@ interface Handle {
     suspend fun close()
 }
 
+data class HandleSpec<T : Entity>(
```
Comment (jblebrun) [inline java/arcs/core/entity/Handle.kt]: What if this were a `sealed class HandleSpec` with subclasses `SignletonHandleSpec` and `CollectionHandleSpec`? Then the constructors could only accept the properly-typed spec, and `init { check...` could be removed.

File: java/arcs/core/host/EntityHandleManager.kt (hunk 1 above)
Comment (csilvestrini) [reply java/arcs/core/host/EntityHandleManager.kt]: Done

File: java/arcs/core/entity/Handle.kt (hunk 2 above)
Comment (csilvestrini) [reply java/arcs/core/entity/Handle.kt]: That's a neat idea, but there's going to be other ways of splitting this class up too (e.g. EntityHandleSpec vs. ReferenceHandleSpec) and we won't be able to do that for all of them. I think it's better to treat them all the same.

**Pull request merged**